src/schema-parser/avro-schema-parser.ts

```diff
@@ -0,0 +1,347 @@
+  return [
+    'application/vnd.apache.avro;version=1.9.0',
+    'application/vnd.apache.avro+json;version=1.9.0',
+    'application/vnd.apache.avro+yaml;version=1.9.0',
+    'application/vnd.apache.avro;version=1.8.2',
+    'application/vnd.apache.avro+json;version=1.8.2',
+    'application/vnd.apache.avro+yaml;version=1.8.2'
+  ];
+};
+
+const BYTES_PATTERN = '^[\u0000-\u00ff]*$';
+const INT_MIN = Math.pow(-2, 31);
+const INT_MAX = Math.pow(2, 31) - 1;
+const LONG_MIN = Math.pow(-2, 63);
+const LONG_MAX = Math.pow(2, 63) - 1;
+
+const typeMappings: { [key: string]: JSONSchema7TypeName } = {
+  null: 'null',
+  boolean: 'boolean',
+  int: 'integer',
+  long: 'integer',
+  float: 'number',
+  double: 'number',
+  bytes: 'string',
+  string: 'string',
+  fixed: 'string',
+  map: 'object',
+  array: 'array',
+  enum: 'string',
+  record: 'object',
+  uuid: 'string',
+};
+
+const commonAttributesMapping = (avroDefinition: AvroSchema, jsonSchema: AsyncAPISchema, isTopLevel: boolean) => {
+  if (avroDefinition.doc) jsonSchema.description = avroDefinition.doc;
+  if (avroDefinition.default !== undefined) jsonSchema.default = avroDefinition.default;
+
+  const fullyQualifiedName = getFullyQualifiedName(avroDefinition);
+  if (isTopLevel && fullyQualifiedName !== undefined) {
+    jsonSchema['x-parser-schema-id'] = fullyQualifiedName;
+  }
+};
+
+function getFullyQualifiedName(avroDefinition: AvroSchema) {
+  let name;
+
+  if (avroDefinition.name) {
+    if (avroDefinition.namespace) {
+      name = `${avroDefinition.namespace}.${avroDefinition.name}`;
+    } else {
+      name = avroDefinition.name;
+    }
+  }
+
+  return name;
+}
+
+/**
+ * Enrich the parent's required attribute with the required record attributes
+ * @param fieldDefinition the actual field definition
+ * @param parentJsonSchema the parent json schema which contains the required property to enrich
+ * @param haveDefaultValue we assure that a required field does not have a default value
+ */
+const requiredAttributesMapping = (fieldDefinition: any, parentJsonSchema: AsyncAPISchema, haveDefaultValue: boolean) => {
+  const isUnionWithNull = Array.isArray(fieldDefinition.type) && fieldDefinition.type.includes('null');
+
+  // we assume that a union type without null and a field without default value is required
+  if (!isUnionWithNull && !haveDefaultValue) {
+    parentJsonSchema.required = parentJsonSchema.required || [];
+    parentJsonSchema.required.push(fieldDefinition.name);
+  }
+};
+
+function extractNonNullableTypeIfNeeded(typeInput: any, jsonSchemaInput: AsyncAPISchema) {
+  let type = typeInput;
+  let jsonSchema = jsonSchemaInput;
+  // Map example to first non-null type
+  if (Array.isArray(typeInput) && typeInput.length > 0) {
+    const pickSecondType = typeInput.length > 1 && typeInput[0] === 'null';
+    type = typeInput[+pickSecondType];
+    if (jsonSchema.oneOf !== undefined) {
+      jsonSchema = jsonSchema.oneOf[0] as AsyncAPISchema; // TODO we lose boolean type?
```

Comment:
What do you mean by `we lose boolean type?`? Do you mean that JSON Schema can be true or false? I don't think so that it's a problem here.